packages/test/test-end-to-end-tests/src/test/data-migration/basicMigration.spec.ts

```diff
@@ -0,0 +1,186 @@
+	rootDataObject: RootDO2,
+	provider: ITestObjectProvider,
+	runtimeFactory: ContainerRuntimeFactoryWithDefaultDataStore,
+	readContainer?: IContainer,
+) {
+	// Validate in memory objects
+	const view = rootDataObject.doWithST.view;
+	const view2 = rootDataObject.doWithST2.view;
+
+	assert.equal(view.root.quantity, lstValue1, "Expected view.quantity to match");
+	assert.equal(view2.root.quantity, lstValue2, "Expected view2.quantity to match");
+	assert.equal(view2.root.dir.value, dirValue, "Expected view2.dir.value to match");
+
+	// Validate handles
+	const dObject = await getObjectByHandle<DOWithST>(rootDataObject, "a");
+	const dObject2 = await getObjectByHandle<DOWithST2>(rootDataObject, "b");
+	const tree = await getObjectByHandle<ITree>(rootDataObject, "tree");
+	const tree1 = await getObjectByHandle<ITree>(dObject, "tree");
+	const tree2 = await getObjectByHandle<ITree>(dObject2, "tree");
+
+	const treeView = tree.viewWith(treeConfig1);
+	const treeView1 = tree1.viewWith(treeConfig1);
+	const treeView2 = tree2.viewWith(treeConfig2);
+
+	assert.equal(treeView.root.quantity, lstValue1, "Expected treeView.quantity to match");
+	assert.equal(treeView1.root.quantity, lstValue1, "Expected treeView1.quantity to match");
+	assert.equal(treeView2.root.quantity, lstValue2, "Expected treeView2.quantity to match");
+	assert.equal(treeView2.root.dir.value, dirValue, "Expected treeView2.dir.value to match");
+
+	// Can send and receive ops
+	const container = readContainer ?? (await provider.loadContainer(runtimeFactory));
+	const rootDataObjectB = (await container.getEntryPoint()) as RootDO2;
+	const viewB = rootDataObjectB.doWithST.view;
+	const view2B = rootDataObjectB.doWithST2.view;
+	await provider.ensureSynchronized();
+
+	treeView.root.quantity = 5;
+	treeView2.root.quantity = 6;
+	treeView2.root.dir.value = "baz";
+	await provider.ensureSynchronized();
+
+	assert.equal(
+		treeView.root.quantity,
+		viewB.root.quantity,
+		"Expected treeView.quantity to match",
+	);
+	assert.equal(treeView.root.quantity, 5, "Expected treeView.quantity to update");
+	assert.equal(
+		treeView2.root.quantity,
+		view2B.root.quantity,
+		"Expected treeView1.quantity to match",
+	);
+	assert.equal(treeView2.root.quantity, 6, "Expected treeView2.quantity to update");
+	assert.equal(
+		treeView2.root.dir.value,
+		view2B.root.dir.value,
+		"Expected treeView2.dir.value to match",
+	);
+	assert.equal(treeView2.root.dir.value, "baz", "Expected treeView2.dir.value to update");
+}
+
+export interface IMigrationStrategy {
+	name: string;
+	runtimeFactory: ContainerRuntimeFactoryWithDefaultDataStore;
+	migrateWithSummary(provider: ITestObjectProvider): Promise<string>;
+	migrateWithoutSummary(provider: ITestObjectProvider): Promise<RootDO2>;
+	migrateWithManyContainers(...containers: IContainer[]): Promise<RootDO2[]>;
+}
+
+const migrationStrategies: IMigrationStrategy[] = [];
+
+describeCompat.skip("basicMigration", "NoCompat", (getTestObjectProvider) => {
```

Comment:
would it make more sense to put these in the local-server or functional test packages, which are easier to use the real types in, as they do not use compat. additionally, they don't need to run against multiple servers, which tends to make the tests ever more complicated. 

Eventually we probably want to run against real services, but might be premature now.

Reply:
Sounds good to me

Reply:
Updated

Reply:
As a side note, I think I would also want to run these against tinylicious as well.